robottelo/hosts.py

```diff
@@ -692,7 +696,20 @@ def register(
             options['force'] = str(force).lower()
 
         self._satellite = target.satellite
-        cmd = target.satellite.cli.HostRegistration.generate_command(options)
+        if auth_username and auth_password:
+            user = target.satellite.cli.User.list({'search': f'login={auth_username}'})
+            if user:
+                register_role = target.satellite.cli.Role.info({'name': 'Register hosts'})
+                target.satellite.cli.User.add_role(
```

Comment:
We  should not add any role to the user here. This should be handled at the test level. We just need to check if the user has necessary permissions to register hosts. 
I don't know if we have such negative scenario to test where a non-admin user tries to register the host with no permissions which should fail but it will pass if we add the role here.

Reply:
As this is general helper to register host, so we should handle assigning required roles or permissions to the user, and I wouldn't care of negative tests as we have no tests, and it can be tested with auth_username and auth_password

Reply:
But we are using this to register the host. It should only accept user and password. Keep the role addition part to the test. We can miss a scenario if we add a role here.

Reply:
this role would be necessary anyway to register hosts, so I don't understand which scenario we might miss here if role is assigned already on user, could you share some example?

Reply:
Ok, as we don't have any such tests using this helper yet, and if user has role added to it so if we add it again here, it'll work as we're not asserting role addition anyway, so this would work.

@shweta83 I think we're good to go with this approch for now, and if we encounter any issues with this later on then we can make it either configurable or we can re-think of the other possibilities, but for now I don't think it as a blocker for this PR

Reply:
Yeah it is not a blocker comment. We will update the helper if something comes in our way. We are good for now. Resolving this conversation.